lib/sparkql/parser_tools.rb

```diff
@@ -151,5 +159,15 @@ def validate_multiple_arguments args
       args.slice!(max_values..-1)
     end
   end
+  
+  def coercible_types type1, type2
+    if DATE_TYPES.include?(type1) && DATE_TYPES.include?(type2)
+      DATE_TYPES.first
+    elsif NUMBER_TYPES.include?(type1) && NUMBER_TYPES.include?(type2)
+      NUMBER_TYPES.first
```

Comment:
afterthought: I realized we want to coercer the less-granular type to the more granular (int -> float, date -> datetime), which you're doing implicity here, but it might be worth a comment--either in this method or above by the constants--noting that order of those arrays matters, and a test or two